src/App.tsx

```diff
@@ -1,93 +1,224 @@
+      setError(ErrorType.RemoveError);
+    }
+
+    setDeletedTodosId([]);
+  };
+
+  const editTodo = async (id: number, newTitle: string) => {
+    const updatedTodos = [...todos].map((todo) => {
+      if (todo.id === id) {
+        todo.title = newTitle;
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodos);
+
+    try {
+        setUpdatedTodoID(prevIds => [...prevIds, id]);
+        await updateTodo(id, { title: newTitle });
+
+    } catch {
+      setError(ErrorType.UpdatedError);
+    }
+
+    setUpdatedTodoID([]);
+  };
+
+  const clearCompleted = async () => {
+    const doneTasks = todos.filter(todo => todo.completed);
+    const doneIds = doneTasks.map((todo) => {
+      return todo.id;
+    });
+
+    setDeletedTodosId(doneIds);
+
+    await Promise.all(doneTasks.map((todo) => deleteTodo(todo.id)));
+
+    const visibleTodos = filtredTodos.filter(todo => {
+      return !todo.completed;
+    });
+
+    setTodos(visibleTodos);
+
+    setDeletedTodosId([]);
+  };
+
+  const hendeleCheckboxChange = async (id: number, completed: boolean) => {
+    try {
+      await updateTodo(id, { completed: !completed });
+
+      const checkedTodoList = todos.map(todo => {
+        if (todo.id === id) {
+          return { ...todo, completed: !completed };
+        }
+
+        return todo;
+      });
+
+      setTodos(checkedTodoList);
+    } catch {
+      setError(ErrorType.UpdatedError);
+    }
+  };
+
+  const completedTodos = useMemo(() => {
+    const finishedTodos = todos.filter(todo => todo.completed);
+
+    return finishedTodos.length;
+  }, [todos]);
+
+  const toggleAll = async () => {
+    const totalTodos = todos.length;
+    const areTodosComplited
+    = todos.filter(todo => todo.completed).length === totalTodos;
+
+    setTodos(todos.map(todo => ({ ...todo, completed: !areTodosComplited })));
+
+    try {
+      const notCompletedTodos = todos.filter(todo => !todo.completed);
+
+      if (notCompletedTodos.length) {
+        notCompletedTodos.forEach(async todo => {
+          updateTodo(todo.id, { completed: !todo.completed });
+        });
+      } else {
+        todos.forEach(async todo => {
+          updateTodo(todo.id, { completed: !todo.completed });
+        });
+      }
+    } catch {
+      setError(ErrorType.UpdatedError);
+    }
+  };
+
+  useEffect(() => {
+    loadTodosFromServer();
+  }, []);
+
   return (
     <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
-
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+
+        <NewTodoField
```

Comment:
```suggestion
      <div className="todoapp__content">
        <NewTodoField
```
Don't leave empty lines between parent and child elements